ext/reflection/php_reflection.c

```diff
@@ -849,7 +849,7 @@ static void _function_string(smart_str *str, zend_function *fptr, zend_class_ent
 	}
 	_function_parameter_string(str, fptr, ZSTR_VAL(param_indent.s));
 	smart_str_free(&param_indent);
-	if (fptr->op_array.fn_flags & ZEND_ACC_HAS_RETURN_TYPE) {
+	if ((fptr->op_array.fn_flags & ZEND_ACC_HAS_RETURN_TYPE) && !ZEND_ARG_TYPE_IS_TENTATIVE(&fptr->common.arg_info[-1])) {
```

Comment:
Hm, I think we should still print the type. Maybe just add a "tenative" indicator somewhere. "Tentative Return" maybe.

Reply:
Yes, I like the idea